server/coordinator/shard_picker.go

```diff
@@ -99,3 +100,63 @@ func copyNodeShardMapping(nodeShardsMapping map[string][]storage.ShardNode) map[
 	}
 	return tempNodeShardMapping
 }
+
+// LeastTableShardPicker selects shards based on the number of tables on the current shard,
+// and always selects the shard with the smallest number of current tables.
+type LeastTableShardPicker struct{}
```

Comment:
With this picker, shall we remove the random picker?

There is no need to expose the implementation, name it as `leastTableShardPicker`.